interview-questions.md

```diff
@@ -8,36 +8,111 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 Researched answer:
 
+ Research shows that my answer may be partially right. Its may be as simple as add_foreign... them use the add_column to add the new foeign key. If its early in the process I found out that you can simply so a db: rollback and that will take you one step back in your application.Once ypu roll back you should be in the migration section.After you are enter back in to your migtations you should be able to create new columns with data.
+
 2. Which RESTful routes must always be passed params? Why?
 
 Your answer:
 
+ Destroy(Delete), Update, Edit, and show. Of the top of my brain I think these 4 restful routes must be passes through  params because they all of these restful routes indicate a specific Id. for example Lets work with Update, when your updating a blog post, your not updation all the blog at once. The user can pick and choose what blog post to update. That is why it is passed with params
+
 Researched answer:
 
+* DESTROY: Deletes a specifi content
+* Update: Updates a specific content
+* Edit: Returns a html element for editing a specific content
+* Show: displays a specific conrent unlike Index
+
+
+
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: 
+
+If I can dig deep into my recollection the three rails generators commands I can think of are called a 
+rails db :migrate, rails db:new and rails db: generate.
+
+We use the rails generate command to create new folders amd files to our applications, It can also be used to unify your work flow. We use the migrate command to change data in you database schema. Migration is a special tool we use to coordinate with our actual code. We use the rails new command to create a new rails application.
 
 Researched answer:
+ 
+ I decided since time is given that I would take the opportunity to look uo some more interesting rails generating commands. Rials server is also known as rails s an it is a command that lets developers work on there application through the command line instead of the GUI, which is a great way to dowmload dependecies like npm and yarn. Another rails generator command that is used often is rails console which is also known as rails c. We use this command to open our developer browser. Which is http://localhost:3000. 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
```

Comment:
looks like you got the correct controller. Just missing what they are used for
https://github.com/learn-academy-2022-echo/Syllabus/blob/main/rails/restful-routes-crud.md